app/src/utc_offset.rs

```diff
@@ -0,0 +1,151 @@
+use chrono::{Duration, TimeZone, Utc};
+use chrono_tz::{OffsetComponents, Tz};
+use std::collections::HashSet;
+use std::ops::RangeInclusive;
+
+pub const TZ_MIN: i64 = -12;
+pub const TZ_MAX: i64 = 14;
+
+#[derive(Debug, Clone)]
+pub struct UtcOffsetRange {
+    primary: RangeInclusive<Duration>,
+    secondary: Option<RangeInclusive<Duration>>,
+    timezone: Duration,
```

Comment:
Badly named variable. This duration is more specifically the offset of the UTC offset

```suggestion
    offset: Duration,
```